app/api/tutorials/[id]/steps/[stepNumber]/route.ts

```
@@ -0,0 +1,178 @@
+        const content = lines.slice(startIdx, endIdx).join('\n');
+        snippets.push({ ...desc, content });
+      } catch (error) {
+        console.warn(`Failed to load snippet from ${filePath}:`, error);
+      }
+    }
+
+    // 1) Parse <CodeFromFiles snippets={[{...}, {...}]}/> blocks
+    // Robust parser that balances braces to extract snippets={[ ... ]}
+    const filesTagRegex = /<CodeFromFiles\s+([^>]*?)\/>/g;
+    let filesTagMatch: RegExpExecArray | null;
+    while ((filesTagMatch = filesTagRegex.exec(parsed.content)) !== null) {
```

Biome (2.1.2)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)